### ISSUE TITLE
org.jooq.util.jaxb.tools.MiniJAXB - org.xml.sax.SAXParseException

### ISSUE THREAD

**Issue by friscoMad** [etiennestuder/gradle-jooq-plugin#165]
Since we did a stack update I am seeing this new warning I have double-checked our configuration but I can't find the cause of the warning, it seems to be a validation exception but the build and generation complete properly.

I am not sure if the problem is in the plugin generated XML or it is thrown in Jooq.

```
> Task :generateJooq
15:15:48.039 [main] WARN org.jooq.util.jaxb.tools.MiniJAXB - org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 5464; cvc-complex-type.2.4.a: Invalid content was found starting with element '{"http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":primaryKeyTypes}'. One of '{"http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":generatedAnnotationDate, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":constructorPropertiesAnnotation, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":constructorPropertiesAnnotationOnPojos, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":constructorPropertiesAnnotationOnRecords, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":pojosAsJavaRecordClasses, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":pojosAsScalaCaseClasses, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":pojosAsKotlinDataClasses, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":jpaVersion, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":globalDomainReferences, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":commentsOnEmbeddables, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":generatedSerialVersionUID, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":maxMembersPerInitialiser, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":newline, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":indentation}' is expected.
15:15:48.170 [main] DEBUG org.jooq.tools.Convert - Jackson is available     
WARNING: An illegal reflective access operation has occurred
cvc-complex-type.2.4.a: Invalid content was found starting with element '{"http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":primaryKeyTypes}'. One of '{"http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":generatedAnnotationDate, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":constructorPropertiesAnnotation, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":constructorPropertiesAnnotationOnPojos, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":constructorPropertiesAnnotationOnRecords, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":pojosAsJavaRecordClasses, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":pojosAsScalaCaseClasses, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":pojosAsKotlinDataClasses, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":jpaVersion, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":globalDomainReferences, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":commentsOnEmbeddables, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":generatedSerialVersionUID, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":maxMembersPerInitialiser, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":newline, "http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd":indentation}' is expected.

WARNING: Illegal reflective access by com.sun.xml.bind.v2.runtime.reflect.opt.Injector (file:/Users/ramiro.aparicio/.gradle/caches/modules-2/files-2.1/com.sun.xml.bind/jaxb-impl/2.3.0.1/2e979dabb3e5e74a0686115075956391a14dece8/jaxb-impl-2.3.0.1.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int)
An illegal reflective access operation has occurred

Illegal reflective access by com.sun.xml.bind.v2.runtime.reflect.opt.Injector (file:/Users/ramiro.aparicio/.gradle/caches/modules-2/files-2.1/com.sun.xml.bind/jaxb-impl/2.3.0.1/2e979dabb3e5e74a0686115075956391a14dece8/jaxb-impl-2.3.0.1.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int)

WARNING: Please consider reporting this to the maintainers of com.sun.xml.bind.v2.runtime.reflect.opt.Injector
Please consider reporting this to the maintainers of com.sun.xml.bind.v2.runtime.reflect.opt.Injector

WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
15:15:48.318 [main] DEBUG org.jooq.tools.Convert - JAXB is available        
Use --illegal-access=warn to enable warnings of further illegal reflective access operations

All illegal access operations will be denied in a future release

15:15:48.326 [main] INFO org.jooq.codegen.GenerationTool - Initialising properties  : /Users/ramiro.aparicio/workspace/shopper-notification-svc/build/tmp/generateJooq/config.xml
15:15:48.327 [main] DEBUG org.jooq.codegen.GenerationTool - Input configuration      : <onError>FAIL</onError><jdbc><driver>org.postgresql.Driver</driver><url>jdbc:postgresql://localhost:5432/shopper_notification_svc</url><user>shopper_notification_svc_user</user><password></password></jdbc><generator><name>org.jooq.meta.postgres.PostgresKotlinGenerator</name><strategy><name>org.jooq.codegen.CustomStrategy</name></strategy><database><name>org.jooq.meta.postgres.PostgresDatabase</name><regexMatchesPartialQualification>true</regexMatchesPartialQualification><sqlMatchesPartialQualification>true</sqlMatchesPartialQualification><includes>.*</includes><excludes></excludes><includeExcludeColumns>false</includeExcludeColumns><includeTables>true</includeTables><includeEmbeddables>true</includeEmbeddables><includeRoutines>true</includeRoutines><includeTriggerRoutines>false</includeTriggerRoutines><includePackages>true</includePackages><includePackageRoutines>true</includePackageRoutines><includePackageUDTs>true</includePackageUDTs><includePackageConstants>true</includePackageConstants><includeUDTs>true</includeUDTs><includeDomains>true</includeDomains><includeSequences>true</includeSequences><includeIndexes>true</includeIndexes><includePrimaryKeys>true</includePrimaryKeys><includeUniqueKeys>true</includeUniqueKeys><includeForeignKeys>true</includeForeignKeys><includeCheckConstraints>true</includeCheckConstraints><includeSystemIndexes>false</includeSystemIndexes><includeSystemCheckConstraints>false</includeSystemCheckConstraints><includeInvisibleColumns>true</includeInvisibleColumns><recordVersionFields></recordVersionFields><recordTimestampFields></recordTimestampFields><syntheticIdentities></syntheticIdentities><syntheticPrimaryKeys></syntheticPrimaryKeys><overridePrimaryKeys></overridePrimaryKeys><dateAsTimestamp>false</dateAsTimestamp><ignoreProcedureReturnValues>false</ignoreProcedureReturnValues><unsignedTypes>true</unsignedTypes><integerDisplayWidths>true</integerDisplayWidths><inputCatalog></inputCatalog><outputCatalogToDefault>false</outputCatalogToDefault><inputSchema>public</inputSchema><outputSchemaToDefault>false</outputSchemaToDefault><schemaVersionProvider></schemaVersionProvider><catalogVersionProvider></catalogVersionProvider><orderProvider></orderProvider><forceIntegerTypesOnZeroScaleDecimals>true</forceIntegerTypesOnZeroScaleDecimals><logSlowQueriesAfterSeconds>5</logSlowQueriesAfterSeconds><logSlowResultsAfterSeconds>5</logSlowResultsAfterSeconds></database><generate><indexes>true</indexes><relations>true</relations><sequenceFlags>true</sequenceFlags><implicitJoinPathsToOne>true</implicitJoinPathsToOne><deprecated>true</deprecated><deprecationOnUnknownTypes>true</deprecationOnUnknownTypes><instanceFields>true</instanceFields><generatedAnnotation>false</generatedAnnotation><generatedAnnotationType>DETECT_FROM_JDK</generatedAnnotationType><generatedAnnotationDate>false</generatedAnnotationDate><nonnullAnnotation>false</nonnullAnnotation><nonnullAnnotationType>javax.annotation.Nonnull</nonnullAnnotationType><nullableAnnotation>false</nullableAnnotation><nullableAnnotationType>javax.annotation.Nullable</nullableAnnotationType><constructorPropertiesAnnotation>false</constructorPropertiesAnnotation><routines>true</routines><sequences>true</sequences><udts>true</udts><queues>true</queues><links>true</links><keys>true</keys><tables>true</tables><embeddables>true</embeddables><records>true</records><recordsImplementingRecordN>true</recordsImplementingRecordN><pojos>false</pojos><pojosEqualsAndHashCode>false</pojosEqualsAndHashCode><pojosToString>true</pojosToString><pojosAsJavaRecordClasses>false</pojosAsJavaRecordClasses><pojosAsScalaCaseClasses>true</pojosAsScalaCaseClasses><pojosAsKotlinDataClasses>true</pojosAsKotlinDataClasses><immutablePojos>false</immutablePojos><serializablePojos>true</serializablePojos><interfaces>false</interfaces><immutableInterfaces>false</immutableInterfaces><serializableInterfaces>true</serializableInterfaces><daos>false</daos><jpaAnnotations>false</jpaAnnotations><validationAnnotations>false</validationAnnotations><springAnnotations>false</springAnnotations><globalObjectReferences>true</globalObjectReferences><globalCatalogReferences>true</globalCatalogReferences><globalSchemaReferences>true</globalSchemaReferences><globalDomainReferences>true</globalDomainReferences><globalTableReferences>true</globalTableReferences><globalSequenceReferences>true</globalSequenceReferences><globalUDTReferences>true</globalUDTReferences><globalRoutineReferences>true</globalRoutineReferences><globalQueueReferences>true</globalQueueReferences><globalLinkReferences>true</globalLinkReferences><globalKeyReferences>true</globalKeyReferences><globalIndexReferences>true</globalIndexReferences><javadoc>true</javadoc><comments>true</comments><commentsOnCatalogs>true</commentsOnCatalogs><commentsOnSchemas>true</commentsOnSchemas><commentsOnTables>true</commentsOnTables><commentsOnColumns>true</commentsOnColumns><commentsOnEmbeddables>true</commentsOnEmbeddables><commentsOnUDTs>true</commentsOnUDTs><commentsOnAttributes>true</commentsOnAttributes><commentsOnPackages>true</commentsOnPackages><commentsOnRoutines>true</commentsOnRoutines><commentsOnParameters>true</commentsOnParameters><commentsOnSequences>true</commentsOnSequences><commentsOnLinks>true</commentsOnLinks><commentsOnQueues>true</commentsOnQueues><commentsOnKeys>true</commentsOnKeys><sources>true</sources><sourcesOnViews>true</sourcesOnViews><fluentSetters>false</fluentSetters><javaBeansGettersAndSetters>false</javaBeansGettersAndSetters><varargSetters>false</varargSetters><generatedSerialVersionUID>CONSTANT</generatedSerialVersionUID><maxMembersPerInitialiser>500</maxMembersPerInitialiser><fullyQualifiedTypes></fullyQualifiedTypes><emptyCatalogs>false</emptyCatalogs><emptySchemas>false</emptySchemas><javaTimeTypes>true</javaTimeTypes><newline>\n</newline></generate><target><packageName>com.returnly.core.model.db</packageName><directory>/Users/ramiro.aparicio/workspace/shopper-notification-svc/build/generated-src/jooq/main</directory><encoding>UTF-8</encoding><clean>true</clean></target></generator>
```

Plugin version: 5.2
Jooq version: 1.14.4 (tested with 1.14.7 with the same result).
Gradle version: 6.7.1
Java version: 11.0.9

Jooq gradle configuration:
```gradle
jooq {
    version = jooq_version
    configurations {
        main {
            generationTool {
                jdbc {
                    driver = "org.postgresql.Driver"
                    url = db_url
                    user = db_user
                    password = ""
                }
                generator {
                    name = "org.jooq.meta.postgres.PostgresKotlinGenerator"
                    strategy {
                        name = "org.jooq.codegen.CustomStrategy"
                    }
                    database {
                        name = "org.jooq.meta.postgres.PostgresDatabase"
                        inputSchema = db_schema
                        includes = ".*"
                        excludes = "flyway_schema_history"
                        includeExcludeColumns = true
                        unsignedTypes = false
                    }
                    generate {
                        fluentSetters = true
                        immutablePojos = true
                        validationAnnotations = true
                        javaTimeTypes = true
                        routines = false
                        indexes = false
                        sequences = false
                        relations = false
                    }
                    target {
                        packageName = "com.returnly.core.model.db"
                    }
                }
            }
        }
    }
}
```
We are using a custom generator class but the warning is thrown before it is instantiated.

**Comment by etiennestuder** [comment]
> Jooq version: 1.14.4 (tested with 1.14.7 with the same result).

That looks like a very old jOOQ version. See [here](https://www.jooq.org/download/).

**Comment by etiennestuder** [comment]
Please try:
1. set jOOQ version to 3.14.7 (see [here](https://github.com/etiennestuder/gradle-jooq-plugin/blob/master/README.md#gradle-groovy-dsl-2))
2. set the latest jOOQ schema version 3.14.7(see [here](https://github.com/etiennestuder/gradle-jooq-plugin/blob/master/README.md#gradle-groovy-dsl-3))

**Comment by etiennestuder** [comment]
@friscoMad I was able to reproduce your behavior and my [suggestion above](https://github.com/etiennestuder/gradle-jooq-plugin/issues/165#issuecomment-778256525) indeed solves the issue.

**Comment by friscoMad** [comment]
Thanks, the version was wrong, we are using 3.14.4, and in fact, setting the schema to 3.14.X does fix the issue.

**Comment by etiennestuder** [comment]
FYI, I published 5.2.1. It will be available on the plugin portal in the coming hours. You can then remove the workaround.
